addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -86,7 +86,8 @@ def _compute_partner_id(self):
 
     @api.depends('task_id', 'task_id.project_id')
     def _compute_project_id(self):
-        for line in self.filtered(lambda line: not line.project_id):
+        for line in self.filtered(lambda line: not line.project_id or (
+                    line.task_id.project_id and line.project_id != line.task_id.project_id)):
```

Comment:
I think this is equivalent to:

```suggestion
        for line in self.filtered(lambda line: line.task_id.project_id and line.project_id != line.task_id.project_id):
```
since if `line.project_id` is falsy, it will be different than a non-falsy `line.task_id.project_id`